arcaflow_plugin_fio/fio_schema.py

```diff
@@ -58,108 +58,357 @@ def is_sync(self) -> bool:
+            "size range instead of a fixed value, the smallest block size is used as "
+            "the metric. Comma-separated values may be specified for reads, writes, "
+            "and trims as described in blocksize."
+        ),
+    ] = None
+    rate_process: typing.Annotated[
+        typing.Optional[RateProcess],
+        schema.name("Rate Process"),
+        schema.description(
+            "Controls how fio manages rated I/O submissions. The default is 'linear', "
+            "which submits I/O in a linear fashion with fixed delays between I/Os that "
+            "gets adjusted based on I/O completion rates. If this is set to 'poisson', "
+            "fio will submit I/O based on a more real world random request flow, known "
+            "as the Poisson process "
+            "(https://en.wikipedia.org/wiki/Poisson_point_process). The lambda will be "
+            "10^6 / IOPS for the given workload."
+        ),
+    ] = None
+
+    # Threads, processes, and job synchronization
+    stonewall: typing.Annotated[
+        typing.Optional[int],
         validation.min(0),
         validation.max(1),
-    ] = field(
-        default=0,
-        metadata={
-            "name": "Atomic",
-            "description": "attemp to use atomic direct IO",
-        },
-    )
-    readwrite: Optional[IoPattern] = field(
-        default=IoPattern.read,
-        metadata={
-            "name": "Read/Write",
-            "description": "type of IO pattern",
-        },
-    )
-    rate_process: Optional[RateProcess] = field(
-        default=RateProcess.linear,
-        metadata={
-            "name": "Rate Process",
-            "description": (
-                "Controls the distribution of delay between IO submissions."
-            ),
-        },
-    )
+        schema.name("Stonewall"),
+        schema.description(
+            "Wait for preceding jobs in the job file to exit, before starting this "
+            "one. Can be used to insert serialization points in the job file. A "
+            "stonewall also implies starting a new reporting group, (see "
+            "group_reporting)."
+        ),
+    ] = None
 
 
 @dataclass
 class FioJob:
-    name: Annotated[str, validation.min(1)] = field(
-        metadata={
-            "name": "Name",
-            "description": "The name of the fio job.",
-        }
-    )
-    params: JobParams = field(
-        metadata={
-            "name": "Fio Job Parameters",
-            "description": "Parameters to execute one fio job.",
-        }
-    )
-    cleanup: bool = field(
-        default=True,
-        metadata={
-            "name": "Cleanup",
-            "description": "Cleanup temporary files created during execution.",
-        },
-    )
+    name: typing.Annotated[
+        str,
+        validation.min(1),
+        schema.name("Job Name"),
+        schema.description("User-defined ASCII name of the job."),
+    ]
```

Comment:
Does the job name really have to be ASCII?  ISO-Latin-1 isn't acceptable?  Nor UTF-8?

Also, the `min(1)` is curious:  it seems like this should be implied by the fact that the type is `str` and _not_ `Optional`.